docarray/array/array_stacked.py

```diff
@@ -110,12 +113,17 @@ def _create_columns(
         for field_to_stack, to_stack in columns_to_stack.items():
 
             type_ = cls.document_type.__fields__[field_to_stack].type_
-            if issubclass(type_, BaseDocument):
-                columns[field_to_stack] = DocumentArrayStacked.__class_getitem__(type_)(
-                    to_stack
-                )
-            else:
-                columns[field_to_stack] = type_.__docarray_stack__(to_stack)
+            if isinstance(type_, type):
+                if issubclass(type_, BaseDocument):
+                    columns[field_to_stack] = DocumentArray.__class_getitem__(type_)(
+                        to_stack
+                    ).stack()
+
+                elif issubclass(type_, (NdArray, TorchTensor)):
+                    columns[field_to_stack] = type_.__docarray_stack__(to_stack)  # type: ignore # noqa: E501
+
+            elif type_ == AnyTensor:  # should be properly handled with union
```

Comment:
I agree with that comment you left ;)
I think we can, at the beginning, do a check for `is_union_type()` (from the typing_inspect lib), and what is currently inside of `if isinstance(type_, type):` can just come after the `else`.
IMO that is cleaner than having the `isinstance(type_, type)` check

Reply:
shit I added this comment to remind me to handle it properly once I rebase on your PR and I forgot to do it